[Docs] add registration/login examples (#12884)
Sylius/Sylius: docs/getting-started-with-sylius/using-api.rst

Using API
=========

> Since sylius 1.8 we allow you to use our new API based on ApiPlatform.

Reviewer: ```suggestion
Since Sylius 1.8 we allow you to use our new API based on the API Platform.
```

        -H 'Content-Type: application/ld+json' \
        -d '{
        "firstName": "shop",
        "lastName": "user",
        "email": "shop.user@example.com",
        "password": "pa$$word",
        "subscribedToNewsletter": true
    }'

If we get response code `204`, it means our customer has been registered successfully.

.. image:: ../_images/sylius_api/api_platform_shop_customer_post.png

Login to the shop
-----------------

Once the shop customer has been registered, we can now create a login request to get the authentication token, which will allow us to use more shop endpoints.
To get this token, lets create a simple login request:

.. code-block:: bash

    curl -X 'POST' \
        'https://master.demo.sylius.com/api/v2/shop/authentication-token' \
        -H 'accept: application/json' \
        -H 'Content-Type: application/json' \
        -d '{
        "email": "shop.user@example.com",
        "password": "pa$$word"
    }'

.. image:: ../_images/sylius_api/api_platform_shop_customer_post.png

As a response we should get code `200`, along with `token` and `customer` iri.

.. code-block:: json

    {
        "token": "string",
        "customer": "iri"
    }

.. note::

>     If your shop has been customized and user first need to authenticate by `email authentication`, the token won't be returned.

Reviewer: ```suggestion
    If your shop has been customized and the user first needs to verify its email, the token won't be returned.
```
